src/main/java/net/snowflake/client/core/BaseSqlInput.java

```diff
@@ -103,4 +112,53 @@ public SQLXML readSQLXML() throws SQLException {
   public RowId readRowId() throws SQLException {
     throw new SnowflakeLoggedFeatureNotSupportedException(session, "readRowId");
   }
+
+  @SnowflakeJdbcInternalApi
+  protected <T> T readObjectOfType(Class<T> type, Object value, SQLInput sqlInput)
+      throws SQLException {
+    if (SQLData.class.isAssignableFrom(type)) {
+      if (sqlInput == null) {
+        return null;
+      } else {
+        SQLData instance = (SQLData) SQLDataCreationHelper.create(type);
+        instance.readSQL(sqlInput, null);
+        return (T) instance;
+      }
+    } else if (Map.class.isAssignableFrom(type)) {
+      if (value == null) {
+        return null;
+      } else {
+        return (T) convertSqlInputToMap((SQLInput) value);
+      }
+    } else if (String.class.isAssignableFrom(type)
+        || Boolean.class.isAssignableFrom(type)
+        || Byte.class.isAssignableFrom(type)
+        || Short.class.isAssignableFrom(type)
+        || Integer.class.isAssignableFrom(type)
+        || Long.class.isAssignableFrom(type)
+        || Float.class.isAssignableFrom(type)
+        || Double.class.isAssignableFrom(type)) {
+      if (value == null) {
+        return null;
+      }
+      return (T) value;
+    } else if (Date.class.isAssignableFrom(type)) {
+      return (T) readDate();
```

Comment:
How come for string we have this value and for date we have to read it? Isn't it breaking our iterator?

Reply:
True, fixed